YahtzeeSim/Score.cs

```diff
@@ -0,0 +1,292 @@
+            int score = 0;
+
+            List<int> playerHand = hand.GetDiceList();
+
+            foreach (int die in playerHand)
+            {
+                if (die == section)
+                {
+                    score += die;
+                }
+            }
+
+            return score;
+        }
+
+        public static int ScoreUpper(UpperSection section, Hand hand)
+        {
+            int score = 0;
+            List<int> playerHand = hand.GetDiceList();
+
+            switch (section)
+            {
+                case UpperSection.threeofakind:
+                    score = ScoreThreeOfAKind(playerHand);
+                    break;
+                case UpperSection.fourofakind:
+                    score = ScoreFourOfAKind(playerHand);
+                    break;
+                case UpperSection.fullhouse:
+                    score = ScoreFullHouse(playerHand);
+                    break;
+                case UpperSection.smallstraight:
+                    score = ScoreSmallStraight(playerHand);
+                    break;
+                case UpperSection.largestraight:
+                    score = ScoreLargeStraight(playerHand);
+                    break;
+                case UpperSection.yahtzee:
+                    score = ScoreYahtzee(playerHand);
+                    break;
+                case UpperSection.chance:
+                    score = ScoreChance(playerHand);
+                    break;
+                default:
+                    Console.WriteLine("Invalid scoring option");
+                    break;
+            }
+
+            return score;
+        }
+
+        public static int ScoreThreeOfAKind(List<int> hand)
+        {
+            if (isThreeOfAKind(hand))
+            {
+                return CountUpDice(hand);
+            }
+            else
+            {
+                return 0;
+            }
+        }
+
+        public static int ScoreFourOfAKind(List<int> hand)
+        {
+            if (isFourOfAKind(hand))
+            {
+                return CountUpDice(hand);
+            }
+            else
+            {
+                return 0;
+            }
+        }
+
+        public static int ScoreFullHouse(List<int> hand)
+        {
+            if (isFullHouse(hand))
+            {
+                return POINTS["fullhouse"];
+            }
+            else 
+            {
+                return 0;
+            }
+        }
+
+        public static int ScoreSmallStraight(List<int> hand)
+        {
+            if (isSmallStraight(hand))
+            {
+                return POINTS["smallstraight"];
+            }
+            else
+            {
+                return 0;
+            }
+        }
+
+        public static int ScoreLargeStraight(List<int> hand)
+        {
+            if (isLargeStraight(hand))
+            {
+                return POINTS["largestraight"];
+            }
+            else
+            {
+                return 0;
+            }

```

Comment:
Do you know about `return isLargeStraight(hand) ? POINTS["largestraight"] : 0` ?